WifiART/src/structs/ArtSystemState.java

```diff
@@ -16,14 +16,35 @@
 
 package structs;
 
-import java.util.ArrayList;
+import java.util.HashMap;
 
 /**
  * The system state enclosing the LCI/LCR parameters.
  */
 public class ArtSystemState {
+
+    // Enum identifying the different subelements.
+    private enum SubelementName {
+        LCI,
+        Z,
+        USAGE,
+        BSSID,
+        LCR,
+        MAP
+    }
+
     // Parameters
-    private ArrayList<Subelement> subelements;
+    private LciState lciState;
+    private ZState zState;
+    private UsageState usageState;
+    private BssidState bssidState;
+    private LcrState lcrState;
+    private MapState mapState;
+    private HashMap<SubelementName, Boolean> includedSubelements;
```

Comment:
Comment on what the mapping is - what does Boolean indicate here?

Reply:
The boolean represents a true/false value of whether the subelement is to be included in the output.